### PR TITLE
front end changes for review comments updates

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
*Issue #:*
https://github.com/bcgov/met-public/issues/1016

*Description of changes:*
- Changes made to handle internal note for any comment review reason3
- Changes made to handle rejection review comments
- Added additional review reason
- Added the option to not send rejection email

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/comments/admin/review/CommentReview.tsx [hunk 1]
```diff
@@ -72,10 +79,20 @@ const CommentReview = () => {
         }
     };
 
+    const extract_staff_note = async () => {
+        staffNote.length === 0
```
Comment (tiagograf-AOT) [inline met-web/src/components/comments/admin/review/CommentReview.tsx:83]: I'd use this the other way around setUpdatedStaffNote(staffNote.len... ? staffNote : default..)

File: met-web/src/constants/commentStatus.ts [hunk 2]
```diff
@@ -2,4 +2,5 @@ export enum CommentStatus {
     Pending = 1,
     Approved = 2,
     Rejected = 3,
+    NeedsFurtherReview = 4,
```
Comment (tiagograf-AOT) [inline met-web/src/constants/commentStatus.ts:5]: If I remember correctly I think NeedsFurtherReview = Pending, so i think we don't need a new status

File: met-web/src/components/comments/admin/review/CommentReview.tsx [hunk 3]
```diff
@@ -295,12 +327,71 @@ const CommentReview = () => {
                                         onChange={(event) => setOtherReason(event.target.value)}
                                     />
                                     <br />
+                                    <MetParagraph>
+                                        <b>Review Note</b> (this note will be inserted in the email sent to the
+                                        respondent to help them understand what needs to be edited for their comment(s)
+                                        to be approved.)
+                                    </MetParagraph>
+                                    {updatedStaffNote.map((staffNote, index) => {
```
Comment (jadmsaadaot) [inline met-web/src/components/comments/admin/review/CommentReview.tsx:335]: Let's create two constant before the return part, const reviewNotes = updatedStaffNote.filter( note => note.type === StaffNoteType[StaffNoteType.Review]. like this then use map to display them here

Comment (jadmsaadaot) [reply met-web/src/components/comments/admin/review/CommentReview.tsx:335]: also what does StaffNoteType[StaffNoteType.Review] return?

Comment (jadmsaadaot) [reply met-web/src/components/comments/admin/review/CommentReview.tsx:335]: consider something like this:
image

**Review by jadmsaadaot — APPROVED** [review]

File: met-web/src/components/comments/admin/review/CommentReview.tsx (hunk 1 above)
Comment (VineetBala-AOT) [reply met-web/src/components/comments/admin/review/CommentReview.tsx]: addressed it, thanks.

File: met-web/src/components/comments/admin/review/CommentReview.tsx (hunk 3 above)
Comment (VineetBala-AOT) [reply met-web/src/components/comments/admin/review/CommentReview.tsx]: addressed it, thanks.

**Review by tiagograf-AOT — APPROVED** [review]

**Pull request merged**